### DISCUSSION TITLE
District heating networks issue (Too many open files) .From URBANopt to Open Modelica

### DISCUSSION THREAD

**Discussion by bvidalEKT** [OpenModelica/OpenModelica#11248] [Q&A] [answered]
Hello, I am conducting simulations of district heating networks, and I have ended up using Open Modelica through URBANopt's DES by NREL, which utilizes the Modelica Building Library 9.1. This is just to say that I have never used Open Modelica before. When I try to run the model, I encounter the following errors.
`[1] 13:16:50 Error
Not possible to open file "C:\Users\Batiste\simulation\Buffalo_Development_DES\run\honeybee_scenario\des_modelica\Loads\Resources\Data\Mixed_use1\modelica.mos" for reading:
Too many open files

[2] 13:16:50 Translation Error
[Buildings.Experimental.DHC.Loads.BaseClasses.getPeakLoad: 79:5-81:23]: [Failed to evaluate function: Buildings.Experimental.DHC.Loads.BaseClasses.getPeakLoad.pureReadLine.](omeditmessagesbrowser:///Buildings.Experimental.DHC.Loads.BaseClasses.getPeakLoad?lineNumber=79)

[3] 13:16:50 Translation Error
[C:/OM121/OM64bit/OMCompiler/Compiler/NFFrontEnd/NFCeval.mo: 1987:20-1987:79]: Internal error NFCeval.evalBuiltinAbs got invalid arguments (TimeSerLoa_e7ef2a44.terUniHea.fan.eff.per.pressure.V_flow[1])`

Is the first one a problem with my computer's capacity? Or could it be related to some limitation imposed by Open Modelica? Regarding the translation issues, I also don't understand how to approach them.

Thank you very much for any hints or forums where I can seek assistance.

**Comment by perost** [comment, marked as answer]
This was fixed in #10775, but it seems the fix didn't make it into the 1.21 release. There should be a new release soon, but you can also try out the nightly build if you don't want to wait.

**Comment by casella** [comment]
A stable release 1.22.0-dev.01 is already available for Linux and will be available soon on Windows. That is recommended instead of the nightly build that changes all the time and could get broken. For the time being on Windows you can use https://build.openmodelica.org/omc/builds/windows/nightly-builds/64bit/older/OpenModelica-v1.22.0-dev-341-gbf24d82543-64bit.exe which is equivalent.

**Reply by bvidalEKT** [reply]
Thank you so much for your rapid response.

I have avoid this problem with a computer with a > 32 Gb RAM.

But when  I run the simulation it crashes at 600 s with the next message:
![image](https://github.com/OpenModelica/OpenModelica/assets/128154169/db046cb5-d266-49fc-8a90-d395c0111db5)

![image](https://github.com/OpenModelica/OpenModelica/assets/128154169/9bc400fb-6f45-442a-9dc4-b23047eb1f70)

**Comment by casella** [comment]
Please try the nightly build. It should also work on the 32 GB machine.
